domain/src/main/kotlin/no/nav/su/se/bakover/domain/vilkår/Vilkårsvurderingsresultat.kt

```diff
@@ -12,43 +12,53 @@ sealed class Vilkårsvurderingsresultat {
 
-        private fun Vilkår.avslagsgrunn(): Avslagsgrunn {
+        private fun Vilkår.avslagsgrunn(): List<Avslagsgrunn> {
             return when (this) {
                 is FastOppholdINorgeVilkår -> {
-                    Avslagsgrunn.BOR_OG_OPPHOLDER_SEG_I_NORGE
+                    listOf(Avslagsgrunn.BOR_OG_OPPHOLDER_SEG_I_NORGE)
                 }
                 is FlyktningVilkår -> {
-                    Avslagsgrunn.FLYKTNING
+                    listOf(Avslagsgrunn.FLYKTNING)
                 }
                 is FormueVilkår -> {
-                    Avslagsgrunn.FORMUE
+                    listOf(Avslagsgrunn.FORMUE)
                 }
                 is InstitusjonsoppholdVilkår -> {
-                    Avslagsgrunn.INNLAGT_PÅ_INSTITUSJON
+                    listOf(Avslagsgrunn.INNLAGT_PÅ_INSTITUSJON)
                 }
                 is LovligOppholdVilkår -> {
-                    Avslagsgrunn.OPPHOLDSTILLATELSE
+                    listOf(Avslagsgrunn.OPPHOLDSTILLATELSE)
                 }
                 is OpplysningspliktVilkår -> {
-                    Avslagsgrunn.MANGLENDE_DOKUMENTASJON
+                    listOf(Avslagsgrunn.MANGLENDE_DOKUMENTASJON)
                 }
                 is PersonligOppmøteVilkår -> {
-                    Avslagsgrunn.PERSONLIG_OPPMØTE
+                    listOf(Avslagsgrunn.PERSONLIG_OPPMØTE)
                 }
                 is UføreVilkår -> {
-                    Avslagsgrunn.UFØRHET
+                    listOf(Avslagsgrunn.UFØRHET)
                 }
                 is UtenlandsoppholdVilkår -> {
-                    Avslagsgrunn.UTENLANDSOPPHOLD_OVER_90_DAGER
+                    listOf(Avslagsgrunn.UTENLANDSOPPHOLD_OVER_90_DAGER)
                 }
                 is FamiliegjenforeningVilkår -> {
-                    Avslagsgrunn.FAMILIEGJENFORENING
+                    listOf(Avslagsgrunn.FAMILIEGJENFORENING)
                 }
                 is PensjonsVilkår -> {
-                    Avslagsgrunn.PENSJON
+                    val avslagsgrunnerPensjonsVilkår = mutableListOf<Avslagsgrunn>()
+                    if (this.grunnlag.any { it.pensjonsopplysninger.søktUtenlandskePensjoner.resultat() == Vurdering.Avslag }) {
+                        avslagsgrunnerPensjonsVilkår.add(Avslagsgrunn.MANGLER_VEDTAK_UTENLANDSKE_PENSJONSORDNINGER)
+                    }
+                    if (this.grunnlag.any { it.pensjonsopplysninger.søktPensjonFolketrygd.resultat() == Vurdering.Avslag }) {
+                        avslagsgrunnerPensjonsVilkår.add(Avslagsgrunn.MANGLER_VEDTAK_ALDERSPENSJON_FOLKETRYGDEN)
+                    }
+                    if (this.grunnlag.any { it.pensjonsopplysninger.søktAndreNorskePensjoner.resultat() == Vurdering.Avslag }) {
+                        avslagsgrunnerPensjonsVilkår.add(Avslagsgrunn.MANGLER_VEDTAK_ANDRE_NORSKE_PENSJONSORDNINGER)
+                    }
+                    avslagsgrunnerPensjonsVilkår
```

Comment:
```suggestion
                   listOfNotNotNull(
                      this.grunnlag.anyOrNull { it.pensjonsopplysninger.søktUtenlandskePensjoner.resultat() == Vurdering.Avslag }?.Avslagsgrunn.MANGLER_VEDTAK_UTENLANDSKE_PENSJONSORDNINGER,
                      this.grunnlag.anyOrNull { it.pensjonsopplysninger.søktPensjonFolketrygd.resultat() == Vurdering.Avslag }?.Avslagsgrunn.MANGLER_VEDTAK_ALDERSPENSJON_FOLKETRYGDEN,
                      this.grunnlag.anyOrNull { it.pensjonsopplysninger.søktAndreNorskePensjoner.resultat() == Vurdering.Avslag }?.Avslagsgrunn.MANGLER_VEDTAK_ANDRE_NORSKE_PENSJONSORDNINGER,
                   )
```
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/list-of-not-null.html